app/registries/menu/manager.rb

```diff
@@ -34,7 +34,7 @@ def items(menu_name)
       end
 
       def to_hash(menu_name)
-        items(menu_name).children.map(&:to_hash)
+        items(menu_name).authorized_children.map(&:to_hash)
```

Comment:
does this work as expected? without reading the code AFAIR this is a registry which might effect all users?

Reply:
`to_hash` is only called when generating the hash for which menus to display afaict

Reply:
would it make sense to create a class instance per user that represent the menus? (in theory might be easy to json (and test) that object vs the menu register)

Reply:
not sure it's worth the effort, it would take a fairly large refactoring to change it and in any case is outside the scope of this issue (which is a 1.20 blocker). right now the `to_hash`, `authorized?` and `authorized_children` methods are called at runtime with a user context and don't modify the registry and can be tested just fine.